bridge/src/worker.rs

```diff
@@ -110,6 +118,47 @@ impl Worker {
         Ok(())
     }
 
+    pub fn poll_workflow_activation<F>(&self, callback: F) -> Result<(), WorkerError> where F: FnOnce(WorkerResult) + Send + 'static {
+        let core_worker = self.core_worker.clone();
+        let callback_tx = self.callback_tx.clone();
+
+        self.tokio_runtime.spawn(async move {
+            let result = core_worker.poll_workflow_activation().await;
+
+            let callback: Callback = match result {
+                Ok(task) => {
+                    let bytes = task.encode_to_vec();
+                    Box::new(move || callback(Ok(bytes)))
+                },
+                Err(PollWfError::ShutDown) => Box::new(move || callback(Err(WorkerError::Shutdown()))),
+                Err(e) => Box::new(move || callback(Err(WorkerError::UnableToPollWorkflowActivation(e))))
+            };
+
+            callback_tx.send(Command::RunCallback(callback)).expect("Unable to send a callback");
+        });
+
+        Ok(())
+    }
+
+    pub fn complete_workflow_activation<F>(&self, bytes: Vec<u8>, callback: F) -> Result<(), WorkerError> where F: FnOnce(WorkerResult) + Send + 'static {
+        let core_worker = self.core_worker.clone();
+        let callback_tx = self.callback_tx.clone();
+        let proto = WorkflowActivationCompletion::decode(&*bytes)?;
+
+        self.tokio_runtime.spawn(async move {
+            let result = core_worker.complete_workflow_activation(proto).await;
+
+            let callback: Callback = match result {
+                Ok(()) => Box::new(move || callback(Ok(vec!()))),
+                Err(e) => Box::new(move || callback(Err(WorkerError::UnableToCompleteWorkflowActivation(e))))
+            };
+
+            callback_tx.send(Command::RunCallback(callback)).expect("Unable to send a callback");
```

Comment:
Just currious... What will happen if ever `callback` raises an exception?

Reply:
I don't think it will have any effect on Rust, as far as it's concerned — the callback was called. We can try this out to make sure this is the case